File: Esercizi ITS/Esercizio7.cs
/*
Esercizio: Scrivere un programma che, data una stringa, restituisca un conteggio delle lettere in essa presenti.
(si consiglia di usare un array con 26 posizioni, una per ogni lettera dell'alfabeto, inizializzato tutto a zeri scorrere la stringa
 e incrementare il contatore della lettera corrispondente).
*/

using System;
using System.Text;

namespace Esercizi_ITS
{
    class Esercizio7
    {
        static void Main(string[] args)
        {
            // Richiedi il nome
            Console.Write("Inserisci una stringa: ");
            string stringa = Console.ReadLine();

            // Conta le lettere contenute nella stringa
            int charNumA = Convert.ToInt32('a');
            int[] lettere = { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
            byte[] chars = Encoding.ASCII.GetBytes(stringa);
            for (int i = 0; i < chars.Length; ++i)
            {
                int charNum = Convert.ToInt32(chars[i]);
                if (charNum >= charNumA)
                {
                    lettere[charNum - charNumA]++;
                }
            }

            // Stampo il risultato
            Console.WriteLine("La stringa inserita contiene le seguenti lettere:");
            for (int i = 0; i < lettere.Length; ++i)
            {
                if (lettere[i] > 0)
                {
                    Console.WriteLine(" - " + lettere[i] + " lettere " + Convert.ToChar(i + charNumA));
                }
            }

            // Comando per far mantenere aperta la console in attesa di un enter
            Console.ReadLine();
        }
    }
}

File: NodeJS/esercizio7.js
#!/usr/bin/env node
/*
Esercizio: Scrivere un programma che, data una stringa, restituisca un conteggio delle lettere in essa presenti.
(si consiglia di usare un array con 26 posizioni, una per ogni lettera dell'alfabeto, inizializzato tutto a zeri scorrere la stringa
 e incrementare il contatore della lettera corrispondente).
*/

'use strict';

var readline = require('readline-sync');

var stringa = readline.question("Inserisci una stringa a piacimento: ");

var conteggio = new Array(26);
for (var i = 0; i < 26; ++i) {
	conteggio[i] = 0;
}

var codeA = 'a'.charCodeAt(0);
for (var i = 0; i < stringa.length; ++i) {
	conteggio[stringa.charCodeAt(i) - codeA]++;
}

for (var i = 0; i < 26; ++i) {
	if (conteggio[i] > 0) {
		var lettera = codeA + i;
		console.log("Lettera " + String.fromCharCode(lettera) + " presente " + conteggio[i] + " volte.");
	}
}
